Fix parent select screen

jgrabowski121 opened the pull request:
This PR fixes known issues existing in the `add_arc_screen`
These issues include: 
- Inability to travel back up the parent arc tree using the back button in the `parent_select_screen`
- Utilization of streams belonging to the `arc_view_screen`
- Tasks shown in `parent_select_screen` when only arcs should be shown

Additional fixes: 
- Due Date field was not displaying its value is the user entered a due date, and then selected a parent arc (leaving the add arc screen and the returning upon the parent arc selection)
- UI updates for visual improvements
- Removal of unnecessary commented out code

Note: This PR will help in the development of the Add Task screen and will close issue #15

jgrabowski121 committed:
Update add_arc_screen.dart

Create a function to directly create an instance of `ParentSelectScreen` widget

jgrabowski121 committed:
Update parent_select_screen.dart

UI updates for improved screen navigation. Added (x) icon to close the screen and return the user to the `add_arc_screen`

jgrabowski121 committed:
Update bloc.dart

Create seperate streams/controllers to be used by `parent_select_screen`

jgrabowski121 committed:
Update parent_select_screen.dart

Replece instances of arcViewStream with arcParentSelctViewStream

jgrabowski121 committed:
Update bloc.dart

Add new getChildArcs function to only return an Arcs arc children

jgrabowski121 committed:
Update parent_select_screen.dart

flag on line 59 changed to getChildArcs

jgrabowski121 committed:
Update databaseHelper.dart

Create getChildArcs function to get a list of only an Arcs arc children

jgrabowski121 committed:
Update parent_arc_tile.dart

call blocs `parentSelectInsert` function instead of `arcViewInsert`

jgrabowski121 committed:
Update parent_arc_tile.dart

Change call to update `ParentSelectScreen.currentParent`

jgrabowski121 committed:
Update parent_select_screen.dart

Change flag from `getChildren` to `getChildArcs`

jgrabowski121 committed:
Modify add arc screen

Update hint text to use black font. Fixed Due Date field to show the entered date after a parent is selected

jgrabowski121 committed:
Add func to initialize streams for add arc screen

Create function `initializeAddArcSteams` to allow the stream values to be set to null

jgrabowski121 committed:
Update add_arc_screen.dart

Call `bloc.initializeAddArcStream` when the (x) iconButton is pressed

jgrabowski121 committed:
Update submit button

Submit button updated for visual improvement

jgrabowski121 committed:
Update add_arc_screen.dart

Changed onChanged call to onFieldSubmitted

jgrabowski121 commented:
Note: I have discovered a bug, 
Upon further testing, I noticed that the updates to the `add_arc_screen` field `due date` has been rendered no functional. It will currently reset its value if the user clicks the select parent button. I am currently working on a fix

jgrabowski121 committed:
Remove conditionals from hintText

Remove unneccesary conditionals

jgrabowski121 committed:
Fix Navigator call

Fix navigator call to allow the previos screens state to be maintained

jgrabowski121 committed:
Remove comment

Remove unneccesary commented out code

jrm86 reviewed (approved):
Thanks Justin! I had noticed an error when running this branch earlier today (only had a few moments before a meeting, and didn't have time to comment), but your latest commits have addressed it.👍

jgrabowski121 committed:
Remove drawer menu

Drawer menu removed because this screen requires the user to leave it by pressing the cancel (x) button or submitting to allow the streams to be cleared.

jgrabowski121 commented:
Thank you @jrm86 for the review, just a heads up I just pushed the final commit I was working on that removes the drawer menu because if a user leaves the screen by selecting a screen from the drawer menu, then the values in the streams are not reset. Now I believe the screen is completely done. I also disabled the default back button functionality to ensure the cancel (x) button is pressed to cancel the arc creation therefore ensuring we control what happens with the streams if the creation is canceled. I am on the fence of changing this to a back arrow but I think the cancel button is more appropriate.

MatthewChastain reviewed (approved):
Tested and looks really good! Thanks for doing this work @jgrabowski121! This will also serve as most of the base for a Task Creation Screen if I'm not mistaken.

KevinKelly25 reviewed (approved):
Works well @jgrabowski121. Thank you for the fix
